### ISSUE TITLE
Open file macro

### ISSUE THREAD

**Issue by Gabryxx** [wavoigt/FNIRSI-1013D-WAV-Viewer-in-Excel-VBA#3]
![image](https://github.com/user-attachments/assets/899032ab-b1d4-4823-9942-b1da86d10a16)

invalid use of an object

is it perhaps the fault of the installed VBA or the fact that I do not use Office but use Libre Office?
But anyway I think of developing a project in VB.net (I never liked Excel)
Also because the editor offered by this oscilloscope is really poor.
Bye

**Comment by wavoigt** [comment]
Yes, Libre office macros are not compatible with Excel VBA macros.
Let me know if you got a running VB.net Application.

**Comment by Gabryxx** [comment]
Ok bye

**Comment by luizmac** [comment]
Is there some way to run it on Libre Office Calc?

**Comment by wavoigt** [comment]
You can try to translate the VBA code to LibreOffice Basic scripting language, which is very similar.

**Comment by luizmac** [comment]
I've tried using Copilot. The routine to open the file worked well. But the checkboxes are not working and some values like the probe gains are wrong too. But even with the errors now I can export the values to a csv in order to read it on octave.

I've worked few hours and couldn't correct all the macro yet.

**Comment by wavoigt** [comment]
I converted it to LibreOffice ods with same functionality.

**Comment by wavoigt** [comment]
V1.30: Opens also Data format Peco V0005 and Atlan V0.26d and CSV Export

**Comment by wavoigt** [comment]
V1.40: implemented FFT for CH1

**Comment by Gabryxx** [comment]
You should make a tutorial on how to use your macro (I use Libre Office).
The waveform you provided as an example exceeds the limits of the oscilloscope, and this is incorrect, because the voltage data is incorrect.
With my program developed in .net, I have this screen:

<img width="1207" height="707" alt="Image" src="https://github.com/user-attachments/assets/a697605b-2ddf-4eb4-9a91-1842a919d4c0" />

When sampling a signal, you must make sure it doesn't exceed the vertical limits of the screen, because this oscilloscope saves the screen data, not the signal voltages.
Below is an example of my program with two waves, one of which triggers the CL circuit.

<img width="1212" height="709" alt="Image" src="https://github.com/user-attachments/assets/f5bd068b-ab29-4202-b632-374e0c0ea4c1" />

In my program, the minimum and maximum of the wave are also highlighted, automatically or manually (right mouse), or with the left mouse you can see the value in volts of the wave at that point.
Bye

**Comment by wavoigt** [comment]
Wow, looks amazing! Did you create it in Visual Studio? Whitch language, Basic or C# ?
Do you plan to publish it on Github?

Ok, this might be not the best example because it is overflow, but the voltage readings are correct.
Why do you need a tutorial, the Excel and LibreOffice files should be working out of the box?
If i forgot to mention a configuration in LibreOffice, pls let me know.
The basic code should be easy to read. Maybe there should be some more comments.

**Comment by Gabryxx** [comment]
I tried loading the macro into LibreOffice, I tried changing the name of the file to read, I press the LOAD button and the data displayed remains the same as the first file.

[Debug.zip](https://github.com/user-attachments/files/28877267/Debug.zip)
This is my program
Bye

**Comment by wavoigt** [comment]
In I did not open the .exe file the zip file because of security concerns.
So, if your app does not show the correct graph, it has to do with your data reading function.
As you did not deliver the source code, i could not debug it.
If i open the file with 1013D_DisplayData.ods, it looks like this:
<img width="949" height="672" alt="Image" src="https://github.com/user-attachments/assets/07468b01-befb-41d1-ab54-410ff78fdeab" />

**Comment by Gabryxx** [comment]
The screen capture of 3.wav is this:

<img width="1213" height="707" alt="Image" src="https://github.com/user-attachments/assets/b585f941-5905-4e8a-a06e-ee041fbdf829" />
My program consider only VISIBLE data actually and not min/max of buffer data (that i dont need), and display all 2 channels.

The first wave that you see in your screenshot is when trigger of second channel go up, but in CL circuit, the sine wave is more long in amplitude when trigger go low(and i use fo see resonance frequency in a metal detector circuit).
So in right top corner you can set also the value of capacitor used in CL circuit that need for calculate the value of inductor (you can see in text window "Inductance of CL Circuit= 3,311.728 μH").
Bye
